### PR TITLE
Various VCS host handling improvements

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: downloader/src/main/kotlin/VcsHost.kt [hunk 1]
```diff
@@ -150,7 +150,7 @@ enum class VcsHost(
     /**
      * Return whether this host is applicable for the [url] URI.
      */
-    fun isApplicable(url: URI) = url.host == hostname
+    fun isApplicable(url: URI) = url.host.endsWith(hostname)
```
Comment (fviernau) [inline downloader/src/main/kotlin/VcsHost.kt]: The implementation now can return false positives. Did you consider that?

**Comment by sschuberth** [comment]
> The implementation now can return false positives. Did you consider that?

Yes, but I believe the risk is rather low, and thus acceptable.

**Comment by fviernau** [comment]
> Yes, but I believe the risk is rather low, and thus acceptable.

yup, agree.

**Comment by sschuberth** [comment]
@fviernau, Travis has actually [passed](https://travis-ci.com/heremaps/oss-review-toolkit/builds/150853387), but GitHub seems to have some API issues and does not correctly show the status. So, can you merge regardless, please?

**Comment by fviernau** [comment]
@sschuberth yes - merging according to your comment.

**Pull request merged**